connectionpool.go

```diff
@@ -555,7 +551,10 @@ func (pool *hostConnPool) connect() (err error) {
 
 	// lazily initialize the connPicker when we know the required type
 	pool.initConnPicker(conn)
-	pool.connPicker.Put(conn)
+	if err := pool.connPicker.Put(conn); err != nil {
+		conn.Close()
+		return err
+	}
```

Comment:
When `connect` fails it fails the whole pool.
IMHO errors that `Put` returns are not critical, it is better not to close pool because of them.
Best solution here to just log them:


```suggestion
	if err := pool.connPicker.Put(conn); err != nil {
		conn.Close()
		if debug.Enabled {
			pool.logger.Printf("gocql: pool connection was not added to the pool: %w", err)
		}
	}
```